processing/2_validate_compendia.py

```diff
@@ -0,0 +1,561 @@
+# -
+
+# Since experiments have multiple runs there are duplicated experiment ids in the index
+# We will need to remove these so that the count calculations are accurate
+sample_to_strain_table_full_processed = sample_to_strain_table_full[
+    ~sample_to_strain_table_full.index.duplicated(keep="first")
+]
+
+# +
+# Add study id as a label
+pao1_compendium_ids = list(normalized_pao1_core_encoded_df.index)
+pa14_compendium_ids = list(normalized_pa14_core_encoded_df.index)
+
+normalized_pao1_core_encoded_df["study id"] = list(
+    sample_to_strain_table_full_processed.loc[pao1_compendium_ids, "SRA_study"].values
+)
+normalized_pa14_core_encoded_df["study id"] = list(
+    sample_to_strain_table_full_processed.loc[pa14_compendium_ids, "SRA_study"].values
+)
+# -
+
+normalized_pao1_core_encoded_df.head()
+
+normalized_pa14_core_encoded_df.head()
+
+# +
+# Plot core gene expression in PA14 reference
+fig7 = pn.ggplot(normalized_pao1_core_encoded_df, pn.aes(x="1", y="2"))
+fig7 += pn.geom_point(pn.aes(color="study id"), alpha=0.3)
+fig7 += pn.labs(
+    x="UMAP 1", y="UMAP 2", title="Normalized RNA-seq expression using PA14 reference"
+)
+fig7 += pn.theme_bw()
+fig7 += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=12),
+    axis_title=pn.element_text(family="sans-serif", size=15),
+)
+fig7 += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+print(fig7)
+
+# +
+# Plot core gene expression in PA14 reference
+fig8 = pn.ggplot(normalized_pa14_core_encoded_df, pn.aes(x="1", y="2"))
+fig8 += pn.geom_point(pn.aes(color="study id"), alpha=0.3)
+fig8 += pn.labs(
+    x="UMAP 1", y="UMAP 2", title="Normalized RNA-seq expression using PA14 reference"
+)
+fig8 += pn.theme_bw()
+fig8 += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=12),
+    axis_title=pn.element_text(family="sans-serif", size=15),
+)
+fig8 += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+print(fig8)
+# -
+
+# There are a lot of study ids so its a little hard to see what is happening, but overall it looks like there is clsutering based on study id. I'm still not clear on why there is not more of a separation in the expression data for strain type, but perhaps this will become more clear later.
```

Comment:
Yeah, I think it will be clearer as you dig more in the data, it may not effect your downstream analyses anyways.